tests/client/disco/components/TestInfoDialog.js

```diff
@@ -88,15 +90,15 @@ describe('Clicking outside <InfoDialog />', () => {
           <I18nProvider i18n={getFakeI18nInst()}>
             <div>
               {getInfoDialog()}
-              <div ref="outsideComponent" onClick={(e) => e.stopPropagation()} />
+              <div id="outside-component" onClick={(e) => e.stopPropagation()} />
             </div>
           </I18nProvider>
         );
       }
     }
 
-    const rootComponent = ReactDOM.render(<FakeContainer />, mountNode);
-    const outsideNode = rootComponent.refs.outsideComponent;
```

Comment:
I feel using the callback would be best. The rule specifically mentions this could become async at which point we'll need to start using the callback anyway.

Adding ids to select using `document.querySelector` feels wrong to me since we don't want to be doing that outside of test code.